artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/LargeServerMessageImpl.java

```diff
@@ -298,7 +298,7 @@ public int getBodyBufferSize() {
       } finally {
          if (closeFile) {
             try {
-               file.close();
+               file.close(false);
```

Comment:
these will cause the channel to not flush, correct?

Why you making this change?


The thing is, I;ve run the testsuite after these changes and there are files leaking.

I'm running it one more time but this is the only commit after it started to fail. And I'm a bit confused about this after I merged and looking at the tests.

Reply:
@wy96f  ^^^

Reply:
The getBodyBufferSize only retrieves the fileSize after opening file, and I assume this is only for reading. For this reason, I close it without flushing. I want to ask how you found files leaking? Are the leaking files large message ones?

Reply:
when I run the whole testsuite, the NoProcessFilesBehind is reporting too many files open.